FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/viewmodel/PrivateKeysViewModel.kt

```diff
@@ -210,8 +215,17 @@ class PrivateKeysViewModel(application: Application) : BaseNodeApiViewModel(appl
                 KeyStoreCryptoManager.encryptSuspend(passphrase)
               } else null
 
+            val protectedPrvKey = if (keyDetails.isFullyDecrypted) {
+              PgpKey.encryptKey(
+                requireNotNull(keyDetails.privateKey),
+                Passphrase(requireNotNull(keyDetails.tempPassphrase))
+              )
+            } else {
+              keyDetails.privateKey
+            }
```

Comment:
Here I think you want to check `isFullyEncrypted` because that is what you really care about - that there are no plain packets in the key. If the key was partially encrypted, it would go into `else` and that's a potential vulnerability.

Reply:
Please let me clarify some things. We should support only fully encrypted keys, right? Let's imagine that in that code we receive a key with i`sFullyEncrypted == false` and `passphrase.isNotEmpty() == true`. What should I do?

Reply:
I don't know, that depends on context. Where is the key coming from, and where is the pass phrase coming from, and what is being done?

Reply:
To be able to answer this, please help me by answering these questions:
1) where is the pass phrase coming from here?
2) where is the key coming from here?
3) what is the purpose of this code?

Reply:
> 1. where is the pass phrase coming from here?
> 2. where is the key coming from here?
> 3. what is the purpose of this code?

We use that code every time when we save a private key to the local database: when creating a private key when importing a private key during setup or from the settings. It a single place for all things that relate to saving a key in the local database.

Reply:
What could cause the pass phrase to ever be empty?

Reply:
> What could cause the pass phrase to ever be empty?

Actually, the pass phrase should not be empty here never. We check `passphrase.isNotEmpty() == true` due to Kotlin, because we have the following

![image](https://user-images.githubusercontent.com/2863246/124261637-b1f25680-db39-11eb-8a8e-e496f606dd5a.png)

Reply:
> where is the pass phrase coming from here?

It always comes from the user input.

from here 

![image](https://user-images.githubusercontent.com/2863246/124262003-29c08100-db3a-11eb-86b9-330e52fdb2d1.png)

or

![image](https://user-images.githubusercontent.com/2863246/124262044-393fca00-db3a-11eb-9367-38287d066082.png)

Reply:
In that case, `passphrase.isNotEmpty()` is unrelated to `isFullyEncrypted`, and should be checked separately, independently of one another.

If `passphrase.isEmpty()` then we should throw an exception.
If `keyDetails.isFullyEncrypted != true` for any key, then throw an exception

That means keys submitted into this method must already be fully encrypted by the code that called this method.

I think that's the cleanest approach. But I'm not as familiar with your code structure as you are - if you think this would be not ideal, let me know.

Reply:
I've updated the code to fit your comments